op-node/rollup/finality/finalizer.go

```diff
@@ -243,19 +243,18 @@ func (fi *Finalizer) tryFinalize() {
 	finalizedL2 := fi.lastFinalizedL2 // may be zeroed if nothing was finalized since startup.
 	var finalizedDerivedFrom eth.BlockID
 	// go through the latest inclusion data, and find the last L2 block that was derived from a finalized L1 block
+	fi.log.Debug("tryFinalize", "finalityData", fi.finalityData, "finalizedL2", finalizedL2)
 	for _, fd := range fi.finalityData {
 		if fd.L2Block.Number > finalizedL2.Number && fd.L1Block.Number <= fi.finalizedL1.Number {
 			lastFinalizedBlock := fi.findLastFinalizedL2BlockWithConsecutiveQuorom(fd.L2Block.Number, finalizedL2.Number)
+			fi.log.Debug("finalizing L2 block", "l2_block", fd.L2Block, "derived_from", fd.L1Block, "last_finalized_block", lastFinalizedBlock)
 
 			// set finalized block(s)
-			if lastFinalizedBlock != nil {
-				finalizedL2, finalizedDerivedFrom = fi.updateFinalized(*lastFinalizedBlock, lastFinalizedBlock.L1Origin)
+			if lastFinalizedBlock != nil && lastFinalizedBlock.Number == fd.L2Block.Number {
```

Comment:
this change is gonna break the OP e2e test. any idea how to fix it?

let's make sure we test against OP e2e tests whenever we make changes in the OP codebase

Reply:
In the OP e2e test, FP can't predict how many l2 blocks will be derived, so we need to revert this change.